### PR TITLE
DOCS: Documentation directory structure

### PR DESCRIPTION
**Pull request by cafferata**
Based on an email conversation between me and @tlimoncelli, I'm bringing the documentation directory structure closer to the documentation URLs for maintainability/recognisability.

### CODE REVIEW

**Comment by cafferata** [comment]
Merge conflict with https://github.com/StackExchange/dnscontrol/pull/2879 has been resolved.

File: commands/types/dnscontrol.d.ts [hunk 1]
```diff
@@ -195,7 +195,7 @@ declare const DISABLE_REPEATED_DOMAIN_CHECK: RecordModifier;
  *
  * @see https://docs.dnscontrol.org/language-reference/domain-modifiers/a
  */
-declare function A(name: string, address: string | number, ...modifiers: RecordModifier[]): DomainModifier;
+declare function A(name: string, address: string | number, ...modifiers: RecordModifier[]): ;
```
Comment (cafferata) [inline commands/types/dnscontrol.d.ts:198]: I will look into this deletion (from `go generate ./...`) later.

**Comment by cafferata** [comment]
@tlimoncelli this pull request is ready for review. 👍

**Comment by cafferata** [comment]
Yes! I wanted to double check the failing GitHub action job. 😊

**Comment by tlimoncelli** [comment]
Looking good so far!

One more simplification:

1. Rename the `provider` subdirectory to `service-provider-specific`.

Are you open to breaking old URLs?  I am.  Here are some simplifications that would break old URLs, but would simplify `functionTypes.go`:

2. Don't downcase function names.  `https://docs.dnscontrol.org/language-reference/domain-modifiers/aaaa` would be `https://docs.dnscontrol.org/language-reference/domain-modifiers/AAAA`
3. Don't modify provider names. Why should a user of `CLOUDFLAREAPI` need to know the URL comtains `cloudflare-dns` instead?  `https://docs.dnscontrol.org/service-providers/providers/cloudflare-dns` would be `https://docs.dnscontrol.org/service-providers/providers/CLOUDFLAREAPI` (This also eliminates the need for the `providerNames` variable.

**Comment by cafferata** [comment]
> 1. Rename the `provider` subdirectory to `service-provider-specific`.

That is certainly possible, I would like to split it separately from this GitHub pull request (_for testing purposes._)

> 2. Don't downcase function names

That is GitBook behavior over which we have no control. 🤷🏼‍♂️

> 3. Don't modify provider names.

These names were migrated at the time.

Screenshot 2024-04-09 at 20 02 48

_https://web.archive.org/web/20220624231639/https://stackexchange.github.io/dnscontrol/provider-list_

**Comment by cafferata** [comment]
> Yes! I wanted to double check the failing GitHub action job. 😊

I've applied the `go fmt ./...` through an amend commit and all GitHub actions are now green.
_https://github.com/StackExchange/dnscontrol/pull/2905/checks_

**Comment by cafferata** [comment]
Just applied a rebase on `main`. Ready for review/merge. :+1:

**Review by tlimoncelli — APPROVED** [review]

**Comment by tlimoncelli** [comment]
Looks great! Thanks for doing this big overhaul!

**Pull request merged**